<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>categoryCompare2: Command Line Interface • categoryCompare2</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="categoryCompare2: Command Line Interface">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">categoryCompare2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.100.26</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/command_line_interface.html">categoryCompare2: Command Line Interface</a></li>
    <li><a class="dropdown-item" href="../articles/v2_guide.html">categoryCompare: High-throughput data meta-analysis using gene annotations, V2</a></li>
    <li><a class="dropdown-item" href="../articles/v2_visnetwork_guide.html">categoryCompare2: visNetwork</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>categoryCompare2: Command Line Interface</h1>
            
            <h4 data-toc-skip class="date">2024-10-30
14:58:27.92278</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2/blob/HEAD/vignettes/command_line_interface.Rmd" class="external-link"><code>vignettes/command_line_interface.Rmd</code></a></small>
      <div class="d-none name"><code>command_line_interface.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2" class="external-link">categoryCompare2</a></span><span class="op">)</span></span></code></pre></div>
<p>In addition to using it as part of an R script, there are some
utilities included in <code>{CategoryCompare2}</code> to allow use from
the command line. This is known as a <strong>command line
interface</strong>, or CLI.</p>
<p>Although it still requires having R and the
<a href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2" class="external-link">categoryCompare2</a> package installed, once installed, you
should be able to do a full analysis from the command line.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"moseleybioinformaticslab/categorycompare2"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cli_location</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"exec"</span>, package <span class="op">=</span> <span class="st">"categoryCompare2"</span><span class="op">)</span></span>
<span><span class="va">cli_location</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmpGN9CpH/temp_libpath23d14dd35c6c/categoryCompare2/exec"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="va">cli_location</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "categoryCompare2.R"     "create_annotations.R"   "feature_files_2_json.R"</span></span>
<span><span class="co">#&gt; [4] "filter_and_group.R"     "run_enrichment.R"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>CLI_LOCATION <span class="op">=</span> <span class="va">cli_location</span><span class="op">)</span></span></code></pre></div>
<p>Assuming you are on a linux type system, we can add these to the
path, and set them to be executable. I’m going to set them here in R,
because that’s what I have access to. We show how to do it in the shell
as well.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">old_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"PATH"</span><span class="op">)</span></span>
<span><span class="va">new_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">old_path</span>, <span class="st">":"</span>, <span class="va">cli_location</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>PATH <span class="op">=</span> <span class="va">new_path</span><span class="op">)</span></span></code></pre></div>
<p>Assuming that we’ve saved the path in a shell variable,
<code>CLI_LOCATION</code>, we can add it to the path, and change the
executables to be executable.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="st">"</span><span class="va">$PATH</span><span class="st">:</span><span class="va">$CLI_LOCATION</span><span class="st">"</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">chmod</span> 0750 <span class="va">$CLI_LOCATION</span>/<span class="pp">*</span>.R </span></code></pre></div>
<p>After we make the scripts executable, we can check that we can
actually use it.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">categoryCompare2.R</span> <span class="at">--help</span></span></code></pre></div>
<p>Now, we will create a directory to put our input files and
results.</p>
</div>
<div class="section level2">
<h2 id="inputs">Inputs<a class="anchor" aria-label="anchor" href="#inputs"></a>
</h2>
<p><a href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2" class="external-link">categoryCompare2</a> CLI needs a few different pieces of
information to work:</p>
<ol style="list-style-type: decimal">
<li>Annotations of the features to do enrichment on. An example would be
Gene Ontology terms of gene products.</li>
<li>The set of all features that were measured. For RNA-Seq, this would
be all genes or transcripts in the genome of the organism.</li>
<li>One or more sets of differentially expressed features (genes or
transcripts).</li>
</ol>
<p>For this small example, we are going to use the same estrogen
microarray dataset as in the main vignette. We’ve found the differential
sets of genes for each timepoint, 10 and 48 hours. The set of all genes
measured on the array is in <code>universe_entrez.txt</code>, the 10
hour differential genes are in <code>10_entrez.txt</code>, and the 48
hour differential genes in <code>48_entrez.txt</code>.</p>
</div>
<div class="section level2">
<h2 id="running">Running<a class="anchor" aria-label="anchor" href="#running"></a>
</h2>
<p><a href="https://yihui.org/knitr/" class="external-link">knitr</a> is not good at running sequential bash or shell
commands, so we are going to go through the whole analysis via CLI here,
with some comments, but we will then discuss each one further down, just
without any output.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_loc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test_data"</span>, package <span class="op">=</span> <span class="st">"categoryCompare2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>TESTLOC <span class="op">=</span> <span class="va">test_loc</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">CURR_DIR</span><span class="op">=</span><span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="bu">export</span> <span class="va">WORKING</span><span class="op">=</span><span class="va">$CURR_DIR</span>/cc2_1234</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$WORKING</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$WORKING</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="fu">cp</span> <span class="va">$TESTLOC</span>/10_entrez.txt .</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="fu">cp</span> <span class="va">$TESTLOC</span>/48_entrez.txt .</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="fu">cp</span> <span class="va">$TESTLOC</span>/universe_entrez.txt .</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co"># get the annotations from installed organism database</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="ex">create_annotations.R</span> <span class="at">--orgdb</span><span class="op">=</span>org.Hs.eg.db <span class="at">--feature-type</span><span class="op">=</span>ENTREZID <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  <span class="at">--annotation-type</span><span class="op">=</span>GO <span class="at">--json</span><span class="op">=</span>example_annotations.json</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co"># setup the gene lists</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="ex">feature_files_2_json.R</span> <span class="at">--file1</span><span class="op">=</span>10_entrez.txt <span class="at">--file2</span><span class="op">=</span>48_entrez.txt <span class="dt">\</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>  <span class="at">--universe</span><span class="op">=</span>universe_entrez.txt <span class="at">--json</span><span class="op">=</span>example_features.json</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co"># do the enrichments</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="ex">run_enrichment.R</span> <span class="at">--features</span><span class="op">=</span>example_features.json <span class="dt">\</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>  <span class="at">--annotations</span><span class="op">=</span>example_annotations.json <span class="at">--output-file</span><span class="op">=</span>example_enrichment.txt</span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="co"># filter and find communities of related GO terms by shared feature annotations</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="ex">filter_and_group.R</span> <span class="at">--enrichment-results</span><span class="op">=</span>example_enrichment.txt <span class="dt">\</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>  <span class="at">--p-cutoff</span><span class="op">=</span>0.01 <span class="at">--count-cutoff</span><span class="op">=</span>2 <span class="at">--similarity-file</span><span class="op">=</span>example_similarity.rds <span class="at">--similarity-cutoff</span><span class="op">=</span>0.8 <span class="dt">\</span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>  <span class="at">--table-file</span><span class="op">=</span>example_grouping.txt</span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>  </span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a><span class="co"># cleanup</span></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a><span class="fu">rm</span> <span class="at">-rf</span> <span class="va">$WORKING</span></span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a><span class="co">#&gt;  [H [2J [3JLoading required namespace: GO.db</span></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a><span class="co">#&gt;  [H [2J [3J [H [2J [3J[1] "example_enrichment.txt"</span></span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a><span class="co">#&gt;  [H [2J [3JSignificant Annotations:</span></span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a><span class="co">#&gt; Signficance Cutoffs:</span></span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a><span class="co">#&gt;   counts &gt;= 2</span></span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a><span class="co">#&gt;   padjust &lt;= 0.01</span></span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a><span class="co">#&gt; Counts:</span></span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a><span class="co">#&gt;    10_entrez 48_entrez counts</span></span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a><span class="co">#&gt; G1         1         1    130</span></span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a><span class="co">#&gt; G2         1         0    152</span></span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a><span class="co">#&gt; G3         0         1     42</span></span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a><span class="co">#&gt; G4         0         0  20110</span></span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a><span class="co">#&gt; Saving annotation similarities in: example_similarity.rds</span></span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a><span class="co">#&gt; Removed 17683 edges from graph</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="annotations">Annotations<a class="anchor" aria-label="anchor" href="#annotations"></a>
</h2>
<p>OK, let’s break that down a little more. Annotations are information
about the features. For example, for gene products, we have Gene
Ontology terms, that describe what pathways those products are involved
in (Biological Process), any chemical transformations or binding
properties those gene products might do (Molecular Function), and where
in a cell or other biological structure they might be (Cellular
Component). Other common gene product annotations are also biological
pathway membership like those from Kyoto Encyclopedia of Genes and
Genomes (KEGG), or pathways from Reactome. Similarly, chemical compounds
might be annotated to pathways in KEGG and Reactome.</p>
<p>One of the more common feature annotations are Gene Ontology (GO)
terms. Bioconductor includes GO terms in their organism databases
(org-db), such as <code>org.Hs.eg.db</code>, which is the organism
database for <em>Homo sapiens</em>, indexed by Entrez Gene
(<code>eg</code>).</p>
<p><code>categoryCompare2</code> includes a CLI utility for getting the
GO terms from the included org-db into a form that can be used by the
rest of the CLI, <code>create_annotations.R</code>.</p>
<p>The arguments are:</p>
<ul>
<li>–orgdb: which org-db to use</li>
<li>–feature-type: what type of feature IDs should be used to map to GO
terms</li>
<li>–annotation-type: what annotations to pull out</li>
<li>–json: where the output json should be stored</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">create_annotations.R</span> <span class="at">--orgdb</span><span class="op">=</span>org.Hs.eg.db <span class="at">--feature-type</span><span class="op">=</span>ENTREZID <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="at">--annotation-type</span><span class="op">=</span>GO <span class="at">--json</span><span class="op">=</span>example_annotations.json</span></code></pre></div>
<p>Alternatively, if you have a source of annotations, you can pass that
directly using:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">create_annotations.R</span> <span class="at">--input</span><span class="op">=</span>annotations.json <span class="at">--feature-type</span><span class="op">=</span>ENTREZID <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="at">--annotation-type</span><span class="op">=</span>GO <span class="at">--json</span><span class="op">=</span>example_annotations.json</span></code></pre></div>
<p>For example, the Moseley Bioinformatics Lab has a python project, <a href="https://pypi.org/project/GOcats/" class="external-link"><code>gocats</code></a>, that
enables fuller consideration of term-term relationships in the GO
ontology structure. One of the <code>gocats</code> sub-commands outputs
a json structured file of gene to term mappings that can be used as
input to <code>create_annotations.R</code>,
<code>gocats remap_goterms</code>.</p>
</div>
<div class="section level2">
<h2 id="features">Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h2>
<p>Features are the things we’ve measured. In this example, they are are
genes. They can also be metabolites, chromosomal regions, etc. For
annotation / category enrichment, we need to know what features we
measured (universe), and then what features we are interested in. Part
of the utility of <code>categoryCompare2</code> is providing the ability
to compare the enrichments from multiple lists.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">feature_files_2_json.R</span> <span class="at">--file1</span><span class="op">=</span>10_entrez.txt <span class="at">--file2</span><span class="op">=</span>48_entrez.txt <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="at">--universe</span><span class="op">=</span>universe_entrez.txt <span class="at">--json</span><span class="op">=</span>example_features.json</span></code></pre></div>
<p>In this case, we can combine up to <strong>four</strong> feature
lists using the CLI. If you need to combine more than four feature
lists, you might want to look at the R API directly, or write some code
to create your JSON file directly. The name of the group of features is
taken from the file name. The inputs are:</p>
<ul>
<li>–file1: a set of features (required)</li>
<li>–file2: another set of features (optional)</li>
<li>–file3: more features (optional)</li>
<li>–file4: more features (optional)</li>
<li>–universe: all the features that were measured</li>
<li>–json: the output file</li>
</ul>
</div>
<div class="section level2">
<h2 id="running-enrichments">Running Enrichments<a class="anchor" aria-label="anchor" href="#running-enrichments"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">run_enrichment.R</span> <span class="at">--features</span><span class="op">=</span>example_features.json <span class="dt">\</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>  <span class="at">--annotations</span><span class="op">=</span>example_annotations.json <span class="at">--output-file</span><span class="op">=</span>example_enrichment.txt</span></code></pre></div>
<p>This runs hypergeometric enrichment for each of the feature lists in
the output json file from creating <a href="#features">features</a>
above.</p>
<p>At the very least, there should be the features, annotations, and the
output. A full list of options includes:</p>
<ul>
<li>–config: a YAML configuration file</li>
<li>–default-config: display a default configuration file</li>
<li>–features: the JSON file containing the features (genes) [default:
features.json]</li>
<li>–annotations: the annotations to use, as a file [default:
annotations.json]</li>
<li>–enrichment-test: what type of test to do [default:
hypergeometric]</li>
<li>–enrichment-direction: do you want over- or under-enrichment
[default: over]</li>
<li>–p-adjustment: what kind of p-value correction to perform [default:
BH]</li>
<li>–output-file: Where to save the results [default:
cc2_results.txt]</li>
<li>–text-only: should only the text file be generated? [default:
FALSE]</li>
</ul>
<div class="section level3">
<h3 id="text-only-false">Text Only FALSE<a class="anchor" aria-label="anchor" href="#text-only-false"></a>
</h3>
<p>It’s very important, if you want to do anything further on the CLI
with your results, that you keep <code>--text-only=FALSE</code>. The
next step in the CLI uses an <code>rds</code> file that is generated by
default from the enrichment results. If you use
<code>--text-only=TRUE</code>, then you cannot do <a href="#filter-and-group-annotations">Filter and Group Annotations</a>.
Depending on the analysis you want to do, or your needs, then that may
be fine. Just be aware.</p>
</div>
</div>
<div class="section level2">
<h2 id="filter-and-group-annotations">Filter and Group Annotations<a class="anchor" aria-label="anchor" href="#filter-and-group-annotations"></a>
</h2>
<p>Finally, after running the enrichments, we filter to what we think
should be significant, and alternatively group GO terms by similarity
and look for communities of GO terms.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">filter_and_group.R</span> <span class="at">--enrichment-results</span><span class="op">=</span>example_enrichment.txt <span class="dt">\</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="at">--p-cutoff</span><span class="op">=</span>0.01 <span class="at">--count-cutoff</span><span class="op">=</span>2 <span class="at">--similarity-file</span><span class="op">=</span>example_similarity.rds <span class="dt">\</span></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="at">--similarity-cutoff</span><span class="op">=</span>0.8 <span class="at">--table-file</span><span class="op">=</span>example_grouping.txt</span></code></pre></div>
<p>Although this lists the <code>*.txt</code> file as input, it will
actually be looking for a matching <code>rds</code> file to use, which
has all of the information necessary for the filtering and grouping (see
<a href="#text-only-false">here</a>).</p>
<p>Let’s go through all of the options:</p>
<ul>
<li>–enrichment-results: where the enrichment results are saved
[default: cc2_results.txt]</li>
<li>–p-cutoff: the maximum p-value to consider significant [default:
0.01]</li>
<li>–adjusted-p-values: should adjusted p-values be used if they exist?
[default: TRUE]</li>
<li>–count-cutoff: minimum number of significant features annotated to
an annotation to be considered [default: 2]</li>
<li>–similarity-file: should grouping of annotations be attempted and
saved [default: annotation_similarity.rds]</li>
<li>–similarity-cutoff: minimum similarity measure to consider
annotations linked [default: 0]</li>
<li>–grouping-algorithm: what algorithm should be used to find the
groups [default: walktrap]</li>
<li>–table-file: the results file to save the results [default:
cc2_results_grouped.txt]</li>
<li>–network-file: if desired, save the network as well [default: NULL]
(currently not implemented)</li>
</ul>
<p>If you do grouping, you definitely want to adjust
<code>similarity-cutoff</code> to be something higher, generally, to
make smaller groups of terms. In my own experience, I’ve found 0.8 is a
good value to use for Gene Ontology terms. For other types of
annotations, you may want to use a higher or lower similarity cutoff.
Unfortunately, the outputs you get here will depend on the values of the
<code>p-cutoff</code>, <code>count-cutoff</code>, and
<code>similarity-cutoff</code> used. However, you can iterate on these
fairly rapidly because the enrichment is already done. In addition, the
annotation similarity network is actually saved as an R <code>rds</code>
file, and as long as the same filename is used, instead of recalculating
the annotation similarities, they will be loaded from the
<code>--similarity-file</code>. This also speeds up the computations
considerably.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Robert M Flight.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
