<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>categoryCompare: High-throughput data meta-analysis using gene annotations, V2 â€¢ categoryCompare2</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="categoryCompare: High-throughput data meta-analysis using gene annotations, V2">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">categoryCompare2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.100.32</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/command_line_interface.html">categoryCompare2: Command Line Interface</a></li>
    <li><a class="dropdown-item" href="../articles/gsea.html">Gene Set Enrichment Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/v2_guide.html">categoryCompare: High-throughput data meta-analysis using gene annotations, V2</a></li>
    <li><a class="dropdown-item" href="../articles/v2_visnetwork_guide.html">categoryCompare2: visNetwork</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>categoryCompare: High-throughput data meta-analysis using gene annotations, V2</h1>
                        <h4 data-toc-skip class="author">Robert M
Flight</h4>
            
            <h4 data-toc-skip class="date">2025-06-30
16:21:31.10508</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2/blob/HEAD/vignettes/v2_guide.Rmd" class="external-link"><code>vignettes/v2_guide.Rmd</code></a></small>
      <div class="d-none name"><code>v2_guide.Rmd</code></div>
    </div>

    
    
<!--
% \VignetteEngine{knitr::rmarkdown}
% \VignetteIndexEntry{categoryCompare: High-throughput data meta-analysis using gene annotations, V2}
% \VignetteDepends{categoryCompare2, GO.db, affy, hgu95av2.db, genefilter, estrogen, limma, Cairo, hgu95av2cdf}
% \VignetteKeyword{categoryCompare, meta-analysis, gene ontology, kegg}
% \VignettePackage{categoryCompare2}
-->
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Current high-throughput molecular biology experiments are generating
larger and larger amounts of data. Although there are many different
methods to analyze individual experiments, methods that allow the
comparison of different data sets are sorely lacking. This is important
due to the number of experiments that have been carried out on
biological systems that may be amenable to either fusion or comparison.
Most of the current tools available focus on finding those genes in
experiments that are listed as the same, or that can be shown
statistically that it is significant that the gene was listed in the
results of both experiments.</p>
<p>However, what many of these tools do not do is consider the
similarities (and just as importantly, the differences) between
experimental results at the categorical level. Categorical data includes
any gene annotation, such as Gene Ontologies, KEGG pathways, chromosome
location, etc. categoryCompare has been developed to allow the
comparison of high-throughput experiments at a categorical level, and to
explore those results in an intuitive fashion.</p>
</div>
<div class="section level2">
<h2 id="sample-data">Sample Data<a class="anchor" aria-label="anchor" href="#sample-data"></a>
</h2>
<p>To make the concept more concrete, we will examine data from the
microarray data set <code>estrogen</code> available from Bioconductor.
This data set contains 8 samples, with 2 levels of estrogen therapy
(present vs absent), and two time points (10 and 48 hours). A
pre-processed version of the data is available with this package, the
commands used to generate it are below. Note: the preprocessed one keeps
only the top 100 genes, if you use it the results will be slightly
different than those shown in the vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/affy" class="external-link">"affy"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"hgu95av2.db"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"genefilter"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"estrogen"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">"limma"</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">datadir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"estrogen"</span><span class="op">)</span></span>
<span><span class="va">pd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/read.AnnotatedDataFrame.html" class="external-link">read.AnnotatedDataFrame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">datadir</span>,<span class="st">"estrogen.txt"</span><span class="op">)</span>, </span>
<span>    header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">""</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">pd</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##              estrogen time.h</span></span>
<span><span class="co">## low10-1.cel    absent     10</span></span>
<span><span class="co">## low10-2.cel    absent     10</span></span>
<span><span class="co">## high10-1.cel  present     10</span></span>
<span><span class="co">## high10-2.cel  present     10</span></span>
<span><span class="co">## low48-1.cel    absent     48</span></span>
<span><span class="co">## low48-2.cel    absent     48</span></span>
<span><span class="co">## high48-1.cel  present     48</span></span>
<span><span class="co">## high48-2.cel  present     48</span></span></code></pre>
<p>Here you can see the descriptions for each of the arrays. First, we
will read in the cel files, and then normalize the data using RMA.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">currDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="va">datadir</span><span class="op">)</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/affy/man/read.affybatch.html" class="external-link">ReadAffy</a></span><span class="op">(</span>filenames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">pd</span><span class="op">)</span><span class="op">)</span>, phenoData <span class="op">=</span> <span class="va">pd</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 reading low10-1.cel ...instantiating an AffyBatch (intensity a 409600x8 matrix)...done.</span></span>
<span><span class="co">## Reading in : low10-1.cel</span></span>
<span><span class="co">## Reading in : low10-2.cel</span></span>
<span><span class="co">## Reading in : high10-1.cel</span></span>
<span><span class="co">## Reading in : high10-2.cel</span></span>
<span><span class="co">## Reading in : low48-1.cel</span></span>
<span><span class="co">## Reading in : low48-2.cel</span></span>
<span><span class="co">## Reading in : high48-1.cel</span></span>
<span><span class="co">## Reading in : high48-2.cel</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="va">currDir</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eData</span> <span class="op">&lt;-</span> <span class="fu">affy</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/affy/man/rma.html" class="external-link">rma</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: replacing previous import 'AnnotationDbi::tail' by 'utils::tail' when</span></span>
<span><span class="co">## loading 'hgu95av2cdf'</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'AnnotationDbi::head' by 'utils::head' when</span></span>
<span><span class="co">## loading 'hgu95av2cdf'</span></span></code></pre>
<pre><code><span><span class="co">## Background correcting</span></span>
<span><span class="co">## Normalizing</span></span>
<span><span class="co">## Calculating Expression</span></span></code></pre>
<p>To make it easier to conceptualize, we will split the data up into
two eSet objects by time, and perform all of the manipulations for
calculating significantly differentially expressed genes on each eSet
object.</p>
<p>So for the 10 hour samples:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">e10</span> <span class="op">&lt;-</span> <span class="va">eData</span><span class="op">[</span>, <span class="va">eData</span><span class="op">$</span><span class="va">time.h</span> <span class="op">==</span> <span class="fl">10</span><span class="op">]</span></span>
<span><span class="va">e10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genefilter/man/nsFilter.html" class="external-link">nsFilter</a></span><span class="op">(</span><span class="va">e10</span>, remove.dupEntrez<span class="op">=</span><span class="cn">TRUE</span>, var.filter<span class="op">=</span><span class="cn">FALSE</span>, </span>
<span>        feature.exclude<span class="op">=</span><span class="st">"^AFFX"</span><span class="op">)</span><span class="op">$</span><span class="va">eset</span></span>
<span></span>
<span><span class="va">e10</span><span class="op">$</span><span class="va">estrogen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">e10</span><span class="op">$</span><span class="va">estrogen</span><span class="op">)</span></span>
<span><span class="va">d10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="fl">0</span> <span class="op">+</span> <span class="va">e10</span><span class="op">$</span><span class="va">estrogen</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">d10</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">e10</span><span class="op">$</span><span class="va">estrogen</span><span class="op">)</span></span>
<span><span class="va">fit10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">e10</span>, <span class="va">d10</span><span class="op">)</span></span>
<span><span class="va">c10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">makeContrasts</a></span><span class="op">(</span><span class="va">present</span> <span class="op">-</span> <span class="va">absent</span>, levels<span class="op">=</span><span class="va">d10</span><span class="op">)</span></span>
<span><span class="va">fit10_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html" class="external-link">contrasts.fit</a></span><span class="op">(</span><span class="va">fit10</span>, <span class="va">c10</span><span class="op">)</span></span>
<span><span class="va">eB10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit10_2</span><span class="op">)</span></span>
<span><span class="va">table10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">eB10</span>, number<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">e10</span><span class="op">)</span>, p.value<span class="op">=</span><span class="fl">1</span>, adjust.method<span class="op">=</span><span class="st">"BH"</span><span class="op">)</span></span>
<span><span class="va">table10</span><span class="op">$</span><span class="va">Entrez</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">mget</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">table10</span><span class="op">)</span>, <span class="va">hgu95av2ENTREZID</span>, ifnotfound<span class="op">=</span><span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>And the 48 hour samples we do the same thing:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">e48</span> <span class="op">&lt;-</span> <span class="va">eData</span><span class="op">[</span>, <span class="va">eData</span><span class="op">$</span><span class="va">time.h</span> <span class="op">==</span> <span class="fl">48</span><span class="op">]</span></span>
<span><span class="va">e48</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genefilter/man/nsFilter.html" class="external-link">nsFilter</a></span><span class="op">(</span><span class="va">e48</span>, remove.dupEntrez<span class="op">=</span><span class="cn">TRUE</span>, var.filter<span class="op">=</span><span class="cn">FALSE</span>, </span>
<span>        feature.exclude<span class="op">=</span><span class="st">"^AFFX"</span> <span class="op">)</span><span class="op">$</span><span class="va">eset</span></span>
<span></span>
<span><span class="va">e48</span><span class="op">$</span><span class="va">estrogen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">e48</span><span class="op">$</span><span class="va">estrogen</span><span class="op">)</span></span>
<span><span class="va">d48</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="fl">0</span> <span class="op">+</span> <span class="va">e48</span><span class="op">$</span><span class="va">estrogen</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">d48</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">e48</span><span class="op">$</span><span class="va">estrogen</span><span class="op">)</span></span>
<span><span class="va">fit48</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">e48</span>, <span class="va">d48</span><span class="op">)</span></span>
<span><span class="va">c48</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">makeContrasts</a></span><span class="op">(</span><span class="va">present</span> <span class="op">-</span> <span class="va">absent</span>, levels<span class="op">=</span><span class="va">d48</span><span class="op">)</span></span>
<span><span class="va">fit48_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html" class="external-link">contrasts.fit</a></span><span class="op">(</span><span class="va">fit48</span>, <span class="va">c48</span><span class="op">)</span></span>
<span><span class="va">eB48</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit48_2</span><span class="op">)</span></span>
<span><span class="va">table48</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">eB48</span>, number<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">e48</span><span class="op">)</span>, p.value<span class="op">=</span><span class="fl">1</span>, adjust.method<span class="op">=</span><span class="st">"BH"</span><span class="op">)</span></span>
<span><span class="va">table48</span><span class="op">$</span><span class="va">Entrez</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">mget</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">table48</span><span class="op">)</span>, <span class="va">hgu95av2ENTREZID</span>, ifnotfound<span class="op">=</span><span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>And grab all the genes on the array to have a background set.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gUniverse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">table10</span><span class="op">$</span><span class="va">Entrez</span>, <span class="va">table48</span><span class="op">$</span><span class="va">Entrez</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>For both time points we have generated a list of genes that are
differentially expressed in the present vs absent samples. To compare
the time-points, we could find the common and discordant genes from both
experiments, and then try to interpret those lists. This is commonly
done in many meta-analysis studies that attempt to combine the results
of many different experiments.</p>
<p>An alternative approach, used in <code>categoryCompare</code>, would
be to compare the significantly enriched categories from the two gene
lists. Currently the package supports two category classes, Gene
Ontology, and KEGG pathways. Both are used below.</p>
<p>Note 1: I am not proposing that this is the best way to analyse
<em>this</em> particular data, it is a sample data set that merely
serves to illustrate the functionality of this package. However, there
are many different experiments where this type of approach is definitely
appropriate, and it is up to the user to determine if their data fits
the analytical paradigm advocated here.</p>
</div>
<div class="section level2">
<h2 id="create-gene-list">Create Gene List<a class="anchor" aria-label="anchor" href="#create-gene-list"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/MoseleyBioinformaticsLab/categoryCompare2" class="external-link">"categoryCompare2"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"GO.db"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">table10</span><span class="op">$</span><span class="va">Entrez</span><span class="op">[</span><span class="va">table10</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">g48</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">table48</span><span class="op">$</span><span class="va">Entrez</span><span class="op">[</span><span class="va">table48</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-go-annotation-object">Create GO Annotation Object<a class="anchor" aria-label="anchor" href="#create-go-annotation-object"></a>
</h2>
<p>Before we can do our analysis, we need to define the
<code>annotation</code> object, which maps the annotations to the
features (genes in this case). For a Gene Ontology (GO) based analysis,
this would be <strong>all</strong> the genes annotated to a particular
GO term based on inheritance in the GO DAG. We can generate this list
using the <code>GOALL</code> column of the <code>org.Hs.eg.db</code>,
and then filter to the terms of interest, or use them all.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">go_all_gene</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys <span class="op">=</span> <span class="va">gUniverse</span>, columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GOALL"</span>, <span class="st">"ONTOLOGYALL"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">go_all_gene</span> <span class="op">&lt;-</span> <span class="va">go_all_gene</span><span class="op">[</span><span class="va">go_all_gene</span><span class="op">$</span><span class="va">ONTOLOGYALL</span> <span class="op">==</span> <span class="st">"BP"</span>, <span class="op">]</span></span>
<span><span class="va">bp_2_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">go_all_gene</span><span class="op">$</span><span class="va">ENTREZID</span>, <span class="va">go_all_gene</span><span class="op">$</span><span class="va">GOALL</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bp_2_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">bp_2_gene</span>, <span class="va">unique</span><span class="op">)</span></span>
<span><span class="va">bp_desc</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">GO.db</span>, keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bp_2_gene</span><span class="op">)</span>, columns <span class="op">=</span> <span class="st">"TERM"</span>, keytype <span class="op">=</span> <span class="st">"GOID"</span><span class="op">)</span><span class="op">$</span><span class="va">TERM</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bp_desc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bp_2_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bp_annotation</span> <span class="op">&lt;-</span> <span class="fu">categoryCompare2</span><span class="fu">::</span><span class="fu"><a href="../reference/annotation.html">annotation</a></span><span class="op">(</span>annotation_features <span class="op">=</span> <span class="va">bp_2_gene</span>,</span>
<span>                            description <span class="op">=</span> <span class="va">bp_desc</span>,</span>
<span>                            annotation_type <span class="op">=</span> <span class="st">"GO.BP"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="do-enrichment">Do Enrichment<a class="anchor" aria-label="anchor" href="#do-enrichment"></a>
</h2>
<p>Now we can do hypergeometric enrichment with each of the gene
lists.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g10_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hypergeometric_feature_enrichment.html">hypergeometric_feature_enrichment</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/methods/new.html" class="external-link">new</a></span><span class="op">(</span><span class="st">"hypergeom_features"</span>, significant <span class="op">=</span> <span class="va">g10</span>,</span>
<span>      universe <span class="op">=</span> <span class="va">gUniverse</span>, annotation <span class="op">=</span> <span class="va">bp_annotation</span><span class="op">)</span>,</span>
<span>  p_adjust <span class="op">=</span> <span class="st">"BH"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">g48_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hypergeometric_feature_enrichment.html">hypergeometric_feature_enrichment</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/methods/new.html" class="external-link">new</a></span><span class="op">(</span><span class="st">"hypergeom_features"</span>, significant <span class="op">=</span> <span class="va">g48</span>,</span>
<span>      universe <span class="op">=</span> <span class="va">gUniverse</span>, annotation <span class="op">=</span> <span class="va">bp_annotation</span><span class="op">)</span>,</span>
<span>  p_adjust <span class="op">=</span> <span class="st">"BH"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="combine-and-find-significant">Combine and Find Significant<a class="anchor" aria-label="anchor" href="#combine-and-find-significant"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_combined</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_enrichments.html">combine_enrichments</a></span><span class="op">(</span>g10 <span class="op">=</span> <span class="va">g10_enrich</span>,</span>
<span>                                  g48 <span class="op">=</span> <span class="va">g48_enrich</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_significant_annotations.html">get_significant_annotations</a></span><span class="op">(</span><span class="va">bp_combined</span>, <span class="va">padjust</span> <span class="op">&lt;=</span> <span class="fl">0.001</span>, <span class="va">counts</span> <span class="op">&gt;=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">bp_sig</span><span class="op">@</span><span class="va">statistics</span><span class="op">@</span><span class="va">significant</span></span></code></pre></div>
<pre><code><span><span class="co">## Signficance Cutoffs:</span></span>
<span><span class="co">##   padjust &lt;= 0.001</span></span>
<span><span class="co">##   counts &gt;= 2</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Counts:</span></span>
<span><span class="co">##    g10 g48 counts</span></span>
<span><span class="co">## G1   1   1     72</span></span>
<span><span class="co">## G2   1   0     53</span></span>
<span><span class="co">## G3   0   1     48</span></span>
<span><span class="co">## G4   0   0  14118</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="generate-graph">Generate Graph<a class="anchor" aria-label="anchor" href="#generate-graph"></a>
</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_graph</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_annotation_graph.html">generate_annotation_graph</a></span><span class="op">(</span><span class="va">bp_sig</span><span class="op">)</span></span>
<span><span class="va">bp_graph</span></span></code></pre></div>
<pre><code><span><span class="co">## A cc_graph with</span></span>
<span><span class="co">## Number of Nodes = 135 </span></span>
<span><span class="co">## Number of Edges = 7740 </span></span>
<span><span class="co">##    g10 g48 counts</span></span>
<span><span class="co">## G1   1   1     64</span></span>
<span><span class="co">## G2   1   0     26</span></span>
<span><span class="co">## G3   0   1     45</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_graph</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_edges.html">remove_edges</a></span><span class="op">(</span><span class="va">bp_graph</span>, <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Removed 7530 edges from graph</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_graph</span></span></code></pre></div>
<pre><code><span><span class="co">## A cc_graph with</span></span>
<span><span class="co">## Number of Nodes = 135 </span></span>
<span><span class="co">## Number of Edges = 210 </span></span>
<span><span class="co">##    g10 g48 counts</span></span>
<span><span class="co">## G1   1   1     64</span></span>
<span><span class="co">## G2   1   0     26</span></span>
<span><span class="co">## G3   0   1     45</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_assign</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotation_combinations.html">annotation_combinations</a></span><span class="op">(</span><span class="va">bp_graph</span><span class="op">)</span></span>
<span><span class="va">bp_assign</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assign_colors.html">assign_colors</a></span><span class="op">(</span><span class="va">bp_assign</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="find-communities">Find Communities<a class="anchor" aria-label="anchor" href="#find-communities"></a>
</h3>
<p>It is useful to define the annotations in terms of their
<strong>communities</strong>. To do this we run methods that find and
then label the communities, before generating the visualization and
table.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_communities</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assign_communities.html">assign_communities</a></span><span class="op">(</span><span class="va">bp_graph</span><span class="op">)</span></span>
<span><span class="va">bp_comm_labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/label_communities.html">label_communities</a></span><span class="op">(</span><span class="va">bp_communities</span>, <span class="va">bp_annotation</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="cytoscape-visualization">Cytoscape Visualization<a class="anchor" aria-label="anchor" href="#cytoscape-visualization"></a>
</h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp_vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/vis_in_cytoscape.html">vis_in_cytoscape</a></span><span class="op">(</span><span class="va">bp_graph</span>, <span class="va">bp_assign</span>, <span class="st">"BP"</span><span class="op">)</span></span></code></pre></div>
<p>And we can generate a legend so we know which colors correspond to
which group.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/generate_legend.html">generate_legend</a></span><span class="op">(</span><span class="va">bp_assign</span><span class="op">)</span></span></code></pre></div>
<p><img src="v2_guide_files/figure-html/bp_legend-1.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Robert M Flight.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
